services/business/apps.go

```diff
@@ -97,13 +97,13 @@ func (in *AppService) GetApp(namespace string, app string) (models.App, error) {
 	(*appInstance).Workloads = make([]models.WorkloadSvc, len(appWkd))
 	for i, wkd := range appWkd {
 		wkdSvc := &models.WorkloadSvc{WorkloadName: wkd.Workload}
-		services, _ := in.k8s.GetServicesByDeploymentSelector(namespace, wkd.Deployment)
+		services, _ := in.k8s.GetServices(namespace, wkd.Deployment.Labels)
```

Comment:
Shouldn't it be `wkd.Deployment.Spec.Templace.Labels` ? ie. the labels of the pods that the deployment will create, not the labels of the deployment itself?

Reply:
+1 The deployment labels are meaningless in this context - we need to look at the pod labels - those are what get stored in Prometheus so its the pod labels that we need to look at.

Reply:
+1 absolutely right, it should use the Template labels for this filtering.
Good review.

Reply:
Indeed, I had my notepad open with 
![image](https://user-images.githubusercontent.com/1662329/45556446-90663a00-b83b-11e8-9eec-179d090b533d.png)
Which I forgot to apply.

Good catch :)